src/modules/user/user.middlewares.ts

```diff
@@ -0,0 +1,143 @@
+import { USER_MESSAGES } from '~/modules/user/user.messages'
+import usersService from './user.services'
+import { checkSchema, ParamSchema } from 'express-validator'
+import { validate } from '~/utils/validation'
+
+const usernameSchema: ParamSchema = {
+    notEmpty: {
+        errorMessage: USER_MESSAGES.USERNAME_IS_REQUIRED
+    },
+    isString: {
+        errorMessage: USER_MESSAGES.USERNAME_MUST_BE_STRING
+    },
+    trim: true,
+    isLength: {
+        options: {
+            min: 1,
+            max: 50
+        },
+        errorMessage: USER_MESSAGES.USERNAME_LENGTH_MUST_BE_FROM_1_TO_50
+    }
+}
+
+const emailSchema: ParamSchema = {
+    notEmpty: {
+        errorMessage: USER_MESSAGES.EMAIL_IS_REQUIRED
+    },
+    trim: true,
+    isEmail: {
+        errorMessage: USER_MESSAGES.EMAIL_IS_INVALID
+    }
+}
+
+const phone_numberSchema: ParamSchema = {
+    notEmpty: {
+        errorMessage: USER_MESSAGES.PHONE_NUMBER_IS_REQUIRED
```

Comment:
when register , i have option phone or email, so one of them is optional.
add more logic for "if number phone opt then email require" or "if email opt then number phone required"